### ISSUE TITLE
Reverse proxy issue due to absolute http urls used for image

### ISSUE THREAD

**Issue by andrewleech** [plantuml/plantuml-server#163]
We're using plantuml with gitlab. Recently, anyone using chrome has had issues due to it blocking plantuml images as they're served on http rather than https (viewed on a gitlab page over https).

I'm now trying to run this docker server behind an nginx reverse proxy to resolve this. 

While the basic plantuml web interface is working fine with a simple nginx proxy_pass configuration, it doesn't solve the issue as plantuml still provides the png images on an absolute http url relative to the `proxy_set_header Host` header provided in the nginx configuration.

Is there any way to tell the docker plantuml server to either return png's with a relative url, or an absolute one with https?

**Comment by arnaudroques** [comment]
I'm not 100% sure, but I think the modification to be done is located somewhere here:
https://github.com/plantuml/plantuml-server/blob/master/src/main/webapp/index.jsp#L11

However, I don't know exactly what to do.

Moreover, another solution would be to modify the [dockerfile](https://github.com/plantuml/plantuml-server/blob/master/Dockerfile) to expose https and port 443

**Comment by jarv** [comment]
@andrewleech are you using the `--module=http-forwarded` option? https://github.com/plantuml/plantuml-server/issues/64

In case you are deploying this in GKE GitLab also maintains a helm chart for their own PlantUML deployment that is maintained at https://gitlab.com/gitlab-org/charts/plantuml

**Comment by andrewleech** [comment]
Thanks @arnaudroques and @jarv both your suggestions helped greatly!

In that link to index.jsp#L11, the line above had a great hint `var="scheme"` ... `header['x-forwarded-proto']` which is something I can set in the nginx config.

I've also added that command to the docker run command too, thanks. 

So now, I'm starting plantuml with:
```
docker run -d --name plantuml --restart always --publish 0.0.0.0:8070:8080 plantuml/plantuml-server:jetty --module=http-forwarded
```

and for my reverse proxy:
```
docker run -d --name plantuml_ssl --restart always -v /srv/plantuml/certs:/etc/nginx/certs --publish 0.0.0.0:8074:8074   --entrypoint='/usr/sbin/nginx' fsouza/docker-ssl-proxy:latest

cat << "EOF" >> ./nginx.conf
worker_processes 1;
pid /run/nginx.pid;
daemon off;

error_log stderr info;

events { worker_connections 1024; }

http {
    sendfile on;

    map $http_upgrade $connection_upgrade {
        default upgrade;
        '' close;
    }

    server {
        listen 8074 ssl;

        client_max_body_size 20M;
        ssl_certificate     /etc/nginx/certs/cert.pem;
        ssl_certificate_key /etc/nginx/certs/key.pem;

        add_header Strict-Transport-Security "max-age=31536000" always;

        ssl_session_cache  builtin:1000  shared:SSL:10m;
        ssl_protocols  TLSv1.2 TLSv1.3;
        ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
        ssl_prefer_server_ciphers on;

        location / {
            proxy_pass  http://<docker_host_ip>:8070;
            proxy_redirect     http://myhost.localnet:8070/ https://myhost.localnet:8074/;
            proxy_cache off;

            proxy_set_header Host $host:8074;
            proxy_set_header X-Forwarded-Proto $scheme;
        }
    }
}
EOF

docker cp ./nginx.conf plantuml_ssl:/etc/nginx/nginx.conf
docker restart plantuml_ssl
```

I'm sure there would be a cleaner way to set up the https proxy, but this works! I now have plantuml running perfectly on https

Thanks for the help!

**Comment by SakerGT** [comment]
FWIW, I hit this issue a while back.  My set-up is:
Browser --https--> HAProxy --http--> multiple docker hosts

I start the Jetty-based containers with `--module=http-forwarded` as noted. I created a rule in HAProxy that adds a header `Forwarded: proto=https` per [RFC7239](https://tools.ietf.org/html/rfc7239).  Works great and very simple.

One issue of note: I have encountered a couple of builds of the docker image where `--module=http-forwarded` doesn't work, in which case I don't upgrade.
